tests/dependency/test_dependency_path_management.py

```diff
@@ -0,0 +1,215 @@
+import os
+import random
+import string
+from pathlib import Path
+from typing import Optional
+
+import pytest
+
+from substrafl.constants import TMP_SUBSTRAFL_PREFIX
+from substrafl.dependency import path_management
+
+
+@pytest.fixture
+def src_dir(tmp_path):
+    return tmp_path
+
+
+@pytest.fixture
+def dest_dir(tmp_path):
+    return tmp_path
+
+
+@pytest.fixture
+def create_random_path():
+    def _create_random_path(base_dir: Path, *, path_name_length: int = 10) -> Path:
+        path_name = "".join(random.choices(string.ascii_uppercase + string.digits, k=path_name_length))
+        path = Path(base_dir / path_name)
+        return path
+
+    return _create_random_path
+
+
+@pytest.fixture
+def create_random_file(create_random_path):
+    def _create_random_file(base_dir: Path, *, path_name_length: int = 10, suffix: Optional[str] = None) -> Path:
+        path = create_random_path(base_dir, path_name_length=path_name_length)
+        if suffix:
+            path = path.with_suffix(suffix)
+        path.parent.mkdir(exist_ok=True, parents=True)
+        path.touch()
+        return path
+
+    return _create_random_file
+
+
+@pytest.fixture
+def create_random_folder(create_random_path):
+    def _create_random_folder(base_dir: Path, *, path_name_length: int = 10) -> Path:
+        path = create_random_path(base_dir, path_name_length=path_name_length)
+        path.mkdir()
+        return path
+
+    return _create_random_folder
+
+
+def _get_relative_path(path: Path, src_dir: Path) -> Path:
+    os.chdir(src_dir)
+    return path.relative_to(src_dir.resolve())
+
+
+def test_expand_regex_file(tmp_path, create_random_file):
```

Comment:
do we want to have a test with more than one file? 
same for below with more than one folder / several files in one folder?

Or do we trust that glob is doing its job and do not care about testing that part again?